content/articles/multi-output-classification-with-machine-learning/index.md

```diff
@@ -0,0 +1,315 @@
+From our dataset, we have two output columns: `type` and `rating`. Column distribution is the value count of each column in the entire dataset. We will start with the `type` column.
+
+#### `type` column
+
+```python 
+df['type'].value_counts()
+```
+The output is shown below:
+
+```bash
+Movie      4788
+TV Show    2143
+Name: type, dtype: int64
+```
+In the output above, we have 4788 `movie` data samples and 2143 `TV Show` data samples.
+
+#### `rating` column
+
+To get the value count of the `rating` column, use the following code:
+
+```python
+df['rating'].value_counts()
+```
+The output is shown below:
+
+```bash
+TV-MA    2863
+TV-14    1931
+TV-PG     806
+R         665
+PG-13     386
+TV-Y      280
+Name: rating, dtype: int64
+```
+The output above shows the distribution of all the ratings in our dataset. We have seven ratings: `TV-MA`, `TV-14`, `TV-PG`, `R`, `PG-13` and `TV-Y`.
+
+Before we start building our model, we also need to clean our dataset. Dataset cleaning involves correctly formatting our dataset.
+
+### Text cleaning
+
+For text cleaning, we will convert all our text data into lower case and also remove stop words. We will use the NeatText Python package to perform this process. We will install Neattext using the following code:
+
+```bash
+! pip install neattext
+```
+Let's import the Neattext functions that will be used for text cleaning.
+
+```python
+import neattext.functions as nfx
+```
+To convert the text data into lower case, run this command:
+
+```python
+df['title'] = df['title'].nfx.lower()
+```
+
+Let's remove stopwords from our test dataset. Stopwords are the most common words that are used in any language. They have little weight on the model during training. 
+Removing stopwords removes words with little weight. This allows the model to focus on the words that will have a greater impact during training. 
+To remove the stopwords, run this code:
+
+```python
+df['title'] = df['title'].apply(nfx.remove_stopwords)
+```
+Now that we have removed stop words and correctly formatted our dataset, let's import all the packages that will be building the model.
+
+### Importing important packages
+
+To import all the important packages, run this code:
+
+```python
+from sklearn.linear_model import LogisticRegression
+from sklearn.feature_extraction.text import CountVectorizer
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import accuracy_score
+from sklearn.multioutput import MultiOutputClassifier
+```
+Let's explain these packages that we have imported.
+
+**LogisticRegression**
+This is the algorithm used to train the model. 
+
+**CountVectorizer**
+Since we are dealing with text, we need to convert the input text into vectors of numbers. Machine learning models do not understand the raw text. The converted vectors of numbers are the representation of the original text. CountVectorizer is the most common Python package that is used to perform this process.
```

Comment:
```suggestion
Since we are dealing with text, we need to convert the input text into vectors of numbers. Machine learning models do not understand the raw text. The converted vectors of numbers are the representation of the original text. `CountVectorizer` is the most common Python package used to perform this process.
```